src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -54,18 +54,16 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PropertyGroup>
       <DefaultComposeDir>$(HOME)</DefaultComposeDir>
       <DefaultComposeDir Condition="$(OS.ToUpper().Contains('WINDOWS'))">$(USERPROFILE)</DefaultComposeDir>
-      <DefaultComposeDir>$([System.IO.Path]::Combine($(DefaultComposeDir), '.dotnet', 'packages'))</DefaultComposeDir>
+      <DefaultComposeDir>$([System.IO.Path]::Combine($(DefaultComposeDir), '.dotnet', $(PlatformTarget), 'packages'))</DefaultComposeDir>
+      <ComposeDir Condition="'$(ComposeDir)' != '' and '$(DoNotDecorateComposeDir)' == ''">$([System.IO.Path]::Combine($(ComposeDir), $(PlatformTarget)))</ComposeDir>
       <ComposeDir Condition="'$(ComposeDir)' == ''">$(DefaultComposeDir)</ComposeDir>
-      <ComposeDir Condition="'$(DoNotDecorateComposeDir)' == ''">$(ComposeDir)$(PlatformTarget)\$(_TFM)\</ComposeDir>
+      <ComposeDir Condition="'$(DoNotDecorateComposeDir)' == ''">$([System.IO.Path]::Combine($(ComposeDir), $(_TFM)))</ComposeDir>
       <PublishDir>$([System.IO.Path]::GetFullPath($(ComposeDir)))</PublishDir>
       <_CrossProjFileDir>$([System.IO.Path]::Combine($(MSBuildProjectExtensionsPath),"Optimize"))</_CrossProjFileDir>
       <!-- Ensure any PublishDir has a trailing slash, so it can be concatenated -->
       <PublishDir Condition="!HasTrailingSlash('$(PublishDir)')">$(PublishDir)\</PublishDir>
     </PropertyGroup>
 
-    <RemoveDir
-        Condition="Exists($(PublishDir))"
-        Directories="$(PublishDir)" />
 
     <RemoveDir
```

Comment:
Scratch the "20MB" download part.  That will be cached in the nuget cache.

But even with it cached, deleting this project everytime and re-restoring takes a non-trivial amount of time.